### PR TITLE
Committee Conflict of Interest resolution

### PR DESCRIPTION
**Pull request by justaugustus**
This is a carry of @tpepper's https://github.com/kubernetes/community/pull/6243 / https://github.com/kubernetes/steering/pull/224.

~I still need to sweep the comment/review history from the previous PRs and resolve open concerns, but getting this up ASAP as a point for discussion (as election cycles are already underway).~

Part of https://github.com/kubernetes/steering/issues/221

cc: @kubernetes/steering-committee @jberkus

### CODE REVIEW

File: elections/committee-conflict-of-interest.md
Comment (justaugustus) [inline elections/committee-conflict-of-interest.md:1]: This language reflects the review comments from https://github.com/kubernetes/community/pull/6243#pullrequestreview-812873401.

File: elections/committee-conflict-of-interest.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Committee Conflict of Interest
+
+At time of writing, the only known conflict of interest for serving on
+committees in the Kubernetes Community exists with the
+[Steering Committee](/committee-steering/README.md) and
+the [Code of Conduct Committee](/committee-code-of-conduct/README.md).
+
+**_Kubernetes Community members cannot serve on both committees simultaneously._**
+
+Upon appointment, the (Steering Committee) election committee will reach out to
```
Comment (justaugustus) [inline elections/committee-conflict-of-interest.md:10]: Note here, I've used "Upon appointment...", instead of "Upon nomination...", but I'm happy to update it to something more fitting.

Similar to [maximal representation](https://github.com/kubernetes/steering/blob/c65bd4e586be59d12466b2bd6c19905240a97f0a/elections.md#maximal-representation), the conflict is not triggered until a condition is met i.e., it is determined that the candidate would be serving on both Steering and CoCC.

That condition cannot be met until an election has been completed.

That is to say, if a community member opts to run for:

- either committee while being a standing member of the other, this is allowed
- both committees and is not a standing member of either, that is also allowed

**Review by justaugustus — COMMENTED** [review]
Ready for review.

Committee members:
/cc @kubernetes/steering-committee @kubernetes/code-of-conduct-committee

Election subproject:
/cc @jberkus @parispittman @mrbobbytables @nikhita @fsmunoz
/retitle Committee Conflict of Interest resolution
/hold

File: elections/committee-conflict-of-interest.md (hunk 1 above)
Comment (jberkus) [reply elections/committee-conflict-of-interest.md:10]: for the elections, wouldn't it be better to reach out to the candidate BEFORE the election results are announced, rather than afterwards?

Comment (jberkus) [inline elections/committee-conflict-of-interest.md:10]: This is unclear.

```suggestion
Upon the creation of a potential conflict (after the candidate is nominated for a position that would conflict, and before the conclusion of the election), an uninvolved member of Steering will reach out to them to determine which committee they want to continue with in the event that they are elected.  If the active election is for the Steering Committee, the Election Officers will be informed of this decision, and can request that Steering contact the candidate.
```

Comment (jberkus) [reply elections/committee-conflict-of-interest.md:10]: So, running for both committees is a problem as demonstrated by the current elections.   This puts us in a situation where the results of the election to conclude first need to be put on hold until the second election results are in.  I really think that we need to prohibit running for both positions at once.

**Review by jberkus — CHANGES_REQUESTED** [review]

File: committee-code-of-conduct/election.md [hunk 2]
```diff
@@ -68,19 +74,19 @@ lists, historical data, and any other activities related to the CoCC.
...
 will vote in the special election (ie: eligibility will not be redetermined
-at the time of the special election). A committee member elected in a special
+at the time of the special election). A CoCC member elected in a special
 election will serve out the remainder of the term for the person they are
 replacing, regardless of the length of that remainder.
 
-In the event that the committee votes to dissolve in its entirety an entirely
-new special election will be held by the steering committee instead of filling
+In the event that the CoCC votes to dissolve in its entirety, an entirely
+new special election will he held by the Steering Committee instead of filling
```
Comment (tpepper) [inline committee-code-of-conduct/election.md:89]: Slight typo here: he -> be

File: committee-code-of-conduct/election.md [hunk 3]
```diff
@@ -68,19 +74,19 @@ lists, historical data, and any other activities related to the CoCC.
...
 with names from the previous election. Those elected will complete the remaining
-terms of the previous, dissolved, committee. With those receiving the highest
+terms of the previous, dissolved CoCC's members. Those receiving the highest
```
Comment (tpepper) [inline committee-code-of-conduct/election.md:91]: This sentence fragment reads a bit odd. Maybe “Candidates receiving the highest number of votes will receive the longest term from thise seats open for election.”?

File: committee-code-of-conduct/election.md [hunk 4]
```diff
@@ -44,20 +45,25 @@ Nominations can be collected via any means. In the past a Google form has been u
...
+
+* Vet that all nominees support the role of the Code of Conduct Committee i.e., maintaining a safe and inclusive space for everyone to work on Kubernetes.
 * Confirm with all nominees (that were not self nominations) that they accept the nomination.
 
 ### Maximal representation
 
 Maximal representation of a single employer is `QUORUM - 1`. With a five person committee,
 that limits the number of members employed by the same company to two.
 
+### Committee Conflict of Interest
```
Comment (nikhita) [inline committee-code-of-conduct/election.md:58]: Can we also include a similar note about Conflict of Interest in the steering election docs?

File: elections/committee-conflict-of-interest.md (hunk 1 above)
Comment (nikhita) [reply elections/committee-conflict-of-interest.md]: +1, to update this to `Upon nomination`.

If a candidate is running for both commitees, IMO it is fair to ask them before the election to keep things transparent for everyone.

**Comment by justaugustus** [comment]
Added a section about withdrawing the candidate from the second election to complete in https://github.com/kubernetes/community/pull/7468/commits/b7fd89e0bd8e4dd04af21575170162b44c601627.

File: committee-code-of-conduct/election.md (hunk 4 above)
Comment (justaugustus) [reply committee-code-of-conduct/election.md]: Added in https://github.com/kubernetes/steering/pull/272!

**Review by cblecker — APPROVED** [review]

File: elections/committee-conflict-of-interest.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  committees
+
+Given the uniqueness of the respective committee charters, their memberships'
+position in the community, and the difference in how these committees need to
+adjudicate on community matters, _candidates must be able to declare their
+preference in which committee they would like to serve on, if elected_.
+
+After the candidate is nominated for a position that would conflict and before
+the conclusion of the first of the two elections to complete, a non-conflicted
+Election Officer will reach out to the candidate to determine which committee
```
Comment (jberkus) [inline elections/committee-conflict-of-interest.md:28]:
```suggestion
Election Officer or Steering Committee member will reach out to the candidate to determine which committee
```

... need to cover the case where there isn't a current SC election running.

File: elections/committee-conflict-of-interest.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+they want to continue with in the event that they are elected.
+
+This process should occur as soon as possible after a potential conflict has
+been identified.
+
+If the candidate is elected in the first election to conclude and that
+appointment matches their preference, the candidate will be withdrawn from the
+remaining open election.
```
Comment (jberkus) [inline elections/committee-conflict-of-interest.md:36]: So, what happens if their preference isn't the first election to complete?

**Review by hlipsig — CHANGES_REQUESTED** [review]
Recommend fixing the definition. Subsequent thoughts on dual candidacy worth of larger discussion.

File: committee-code-of-conduct/election.md [hunk 7]
```diff
@@ -68,19 +74,21 @@ lists, historical data, and any other activities related to the CoCC.
 
 ### Vacancies
 
-In the event of a resignation or other loss of an elected committee
+In the event of a resignation or other loss of an elected CoCC
```
[the remark this answers is not in the document]
Comment (justaugustus) [reply committee-code-of-conduct/election.md:77]: @hlipsig -- Thanks for the accessibility callout here!

The first instance of the acronym's usage is actually defined in L3: https://github.com/kubernetes/community/blame/5a77f6ee77da3f7e5dcf0ffc28b430cff266db93/committee-code-of-conduct/election.md#L3

**Review by justaugustus — COMMENTED** [review]
@jberkus @hlipsig -- Thanks for the reviews!

I'll endeavor to address them here.

> So, what happens if their preference isn't the first election to complete?
> What if the first to conclude isn't their preference?

At this stage (presuming the candidate has won a seat in the first election):

1. the candidate would become a member of the first committee's election to complete
2. the _potential_ for a conflict has been identified, but a conflict has not actually occurred

Should the candidate also be elected to a seat for the second committee to complete its election, that is when the conflict is actually realized.

At the conflict stage:

1. The now doubly-elected candidate will have their preference realized i.e., vacate their seat in the first committee
2. The first committee will fill the new vacancy per their election process and/or charter

> Would it be better to just say a candidate cannot run for both committees as the same time?

While the election cycles may have some overlap, they do not truly occur at "the same time" i.e., they do not run for the same duration, they do not start at the same time, and they do not end at the same time.

> I recognize that this is a very unlikely occurrence, but I feel like we should be careful of candidate overlap between candidate pools.

In both this and the above question, my personal opinion is that we should not seek to restrict the opportunities for Kubernetes Community members to run for committee membership outside of what is detailed in the "Eligibility for candidacy" sections of each committee's election process.

---

> Practically, as Josh points outs, allowing candidates to run for both makes for complicated choices. It may even be possible to achieve a deadlock and need to re-run both elections.

@tpepper -- In an attempt to guard against multiple conflicts, I've added the following language:

> In the event that multiple candidates trigger this conflict in an election
cycle, the two (2) affected candidates receiving the most votes will have their
preference honored. In the event that the first election to conclude has only
two seats under consideration, only the affected candidate with the most votes
will have their preference honored.

---

Anecdotally, I want to quickly speak to why making the distinction between the potential for a conflict and an actual conflict is important...

See the following thread: https://groups.google.com/g/kubernetes-dev/c/dXDxb-tOn7Y

Timing of the [maximal representation](https://git.k8s.io/steering/elections.md#maximal-representation) clause's trigger led to a lengthy discussion and varied interpretations between Election Officers and other members of the community **on a candidate's nomination thread**.

If this thread/clause is be considered precedent (as maximal representation is the only conflict we've documented to date), then we cannot consider a conflict to be in effect until all of its conditions have been met.

Explicitly:
A candidate cannot be considered to be in conflict until it is clear that they would be members of both committees.
Their intent to run in both elections does not determine that.
Their winning of a single election does not determine that.

**Comment by katcosgrove** [comment]
Maybe I'm missing some nuance, but if terms for both positions are two years and elections run at different times, I suggest that a candidate is no longer eligible to run in an election for a committee position they do not currently hold if they have won their existing position within the last 12 months. This minimizes the opportunity for conflict, as well as the burden on the community and whichever committee is temporarily disadvantaged by a seat being vacated early.

Example A: Candidate runs for and wins a seat on the CoCC. A few months later, nominations for Steering open. As the candidate won their CoCC seat less than one year ago, they are ineligible to run for Steering.

Example B: Candidate runs for and wins a seat on the CoCC. A year and some change into their term, nominations open for Steering. As the candidate has served more than 12 months on the CoCC, they are eligible to run for Steering. No conflict exists until they win the seat on Steering; if they win, the seat on the CoCC must be vacated, and a special election is triggered for the now-vacant CoCC seat.

Example C: Candidate runs for a seat on the CoCC, but is not successful. They do not currently hold a committee position, so they are eligible to run for Steering that same election year.

It is possible for a candidate holding a seat to willingly vacate their existing position in order to shortcut the 12-month lockout period. For that reason, I also suggest a 3-6 month lockout from running in any election in the case of a seat being vacated voluntarily. If a candidate is experiencing issues in their personal or professional life that prevent them from fulfilling their duties to the Committee rather than vacating the seat simply to get around the lockout period, those issues are unlikely to be resolved in a short period of time.

TL;DR: You shouldn't be able to run for two elections in the same 12-month period if you won the first one.

**Review by cpanato — APPROVED** [review]
/lgtm

thanks for that

**Comment by palnabarun** [comment]
> In both this and the above question, my personal opinion is that we should not seek to restrict the opportunities for Kubernetes Community members to run for committee membership outside of what is detailed in the "Eligibility for candidacy" sections of each committee's election process.

+1 to what @justaugustus said. The concerns about candidates running for multiple committees at different points in time should be addressed in the "Eligibility for Candidacy" sections, for example, a sitting committee member running for another committee.

File: elections/committee-conflict-of-interest.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+If applicable, the committee where that individual vacated their seat will fill
+that vacancy per existing election or committee vacancy rules.
+
+In the event that multiple candidates trigger this conflict in an election
+cycle, the two (2) affected candidates receiving the most votes will have their
+preference honored. In the event that the first election to conclude has only
+two seats under consideration, only the affected candidate with the most votes
+will have their preference honored.
```
Comment (palnabarun) [inline elections/committee-conflict-of-interest.md:47]: > In the event that the first election to conclude has only
two seats under consideration, only the affected candidate with the most votes
will have their preference honored.

The statement sounds like the choice of the candidate second most votes is affected by preference of the candidate with the highest votes. Why should there be a dependence?

Taking an example, let's assume Candidate A and Candidate B have won the elections to Committee 1, but both have prefer to be in Committee 2. Elections to Committee 2 happen in a month and they both intend to run for it.

If they both win the election to Committee 2 and they chose to vacate their seat in Committee 1, the rules for filling vacancy of Committee 1 would be triggered. In my opinion, I don't think in this case, we should restrict the candidate who got the second highest votes in election to Committee 2.

File: elections/committee-conflict-of-interest.md [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+Election Officer or in the event that a Steering Committee election is not
+active, a non-conflicted Steering Committee member will reach out to the
+candidate to determine which committee they want to continue with in the event
+that they are elected.
+
+This process should occur as soon as possible after a potential conflict has
+been identified.
+
+If the candidate is elected in the first election to conclude and that
+appointment matches their preference, the candidate will be withdrawn from the
```
Comment (celestehorgan) [inline elections/committee-conflict-of-interest.md:37]: 👋  Hi everyone, long time no see! 

@justaugustus and I were chatting about this PR casually and I had some thoughts so I figured I'd resurrect myself from the dead, zombie-like, to comment on this briefly. (I'll update my affiliation and email in another PR, then go back to sig docs and hang my head in shame) 

The issue with leaving things up to the candidate's preference is that my instinct tells me that most candidates would prefer a Steering seat over a CoCC one. Steering is higher profile, comes with more power/responsibility and generally "looks better". 

Worded as-is I think we stand to run into an issue where folks run both for CoCC and Steering simultaneously in increased numbers and use CoCC as a "fall back" position, so to speak. There's advantages outside of the project (career-wise, i.e.) to holding committee seats, after all. 

So far, to my knowledge, @parispittman and @tpepper are the only ones who have triggered this particular conflict of interest in practice. In both of their cases, they had served the majority of their Code of Conduct terms (in Paris' case, I think 1 year, and in Tim's 1.5 I believe). 

I would suggest that rather than simply leaving it to preference, Steering **does not allow CoCC members to nominate themselves for SC unless they've completed at least 1 year of their CoCC term**. 

This provides the following benefits:
1. Stability for the CoCC: during the last round of elections we (CoCC) had a newly selected CoCC member put themselves up for SC within 3-4 months of joining the CoCC. It created a bit more flux than we like to see on that committee. Keep in mind that while SC is 10 seats iirc, CoCC is only 5 - one person dropping is 20% of the committee. 
2. Demonstration of commitment: People should not be selected to the CoCC if they aren't willing to commit to the work. While CoCC can provide the community visibility required to run a successful SC candidacy, it shouldn't be viewed as a placeholder by candidates. That cheapens the work and cheapens us as a community. If people aren't willing to commit for at least a year, they shouldn't be putting their name in for the CoCC to begin with 
3. Reduction of conflicts of interest: the real issue isn't a CoCC member being elected to the SC, its **people who run for both simultaneously**. Ideally, people should commit to one or the other. 


Anyways, just my two cents! 

New work is going well and has time for me to do some open source splunking so I'll probably be more around from now on ❤️

**Review by justaugustus — COMMENTED** [review]
@katcosgrove @celestehorgan -- Thanks for kicking around some ideas yesterday! Hopefully I've reasonably represented the discussion in the latest push (https://github.com/kubernetes/community/pull/7468/commits/2c42bb15aa756f592c28b1530623be457438bf9a).

@kubernetes/steering-committee -- PTAL at the latest. I think this solution is much cleaner, as it prioritizes preventing the conflict of interest through eligibility for candidacy, instead of actually having to resolve them after they've been triggered.

File: elections/committee-conflict-of-interest.md [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+- Candidates who are nominated for a seat on both Steering and CoCC within the
+  same 12-month period must:
+  - (if elected) accept a seat on the committee whose election concludes first
+  - be withdrawn from candidacy in second election to conclude
+- Standing Steering or CoCC members must have served on their committee for a
+  period of twelve (12) months to be considered eligible for candidacy in the
+  other committee's election
+- Kubernetes Community members who have recently vacated a seat on either
+  Steering or CoCC will be ineligible for candidacy for a period of three (3)
+  months
```
Comment (justaugustus) [inline elections/committee-conflict-of-interest.md:52]: Reading this bullet over, I wonder if it could be interpreted as disallowing a committee member who is about to complete their term from running for re-election.

If others are interpreting it that way, I'm open to edit suggestions. :)

Comment (katcosgrove) [reply elections/committee-conflict-of-interest.md:52]: This should clear up any potential confusion over your intent here

```suggestion
- Kubernetes Community members who have recently voluntarily vacated a seat on either
  Steering or CoCC without serving the complete term will be ineligible for candidacy for a period of three (3)
  months
```

Comment (jeremyrickard) [reply elections/committee-conflict-of-interest.md:52]: I think the edit from @katcosgrove provides good clarification and captures the intent. 👍

Comment (justaugustus) [reply elections/committee-conflict-of-interest.md]: Neato; thanks Kat!

**Comment by mrbobbytables** [comment]
Overall +1, the one caveat that these should go in effect after our current round of elections conclude as they were not in place before either election was started.

**Comment by justaugustus** [comment]
> Overall +1, the one caveat that these should go in effect after our current round of elections conclude as they were not in place before either election was started.

@mrbobbytables -- Could we instead set an effective date on the top of the doc for Monday, October 2 (Public announcement of Results at Public Steering Committee Meeting)?
ref: https://github.com/kubernetes/community/tree/master/elections/steering/2023#schedule

[This issue](https://github.com/kubernetes/steering/issues/221) has been around for two years and it would be great to not leave this languishing for much longer. 📿

**Comment by justaugustus** [comment]
@mrbobbytables -- okay, perfect. Added an effective date of 2023-10-03 (a day after the election results are announced).

File: elections/committee-conflict-of-interest.md (hunk 8 above)
Comment (palnabarun) [reply elections/committee-conflict-of-interest.md]: Resolving the comment since the block doesn't exist in the latest version.

**Review by palnabarun — APPROVED** [review]
The current version looks good!

Thanks, @justaugustus for working on this.

/lgtm
(with the steering hat)

File: elections/committee-conflict-of-interest.md [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+
+- Loss of committee member that occurs due to this conflict of interest will
+  be resolved per existing election or committee vacancy rules
+- Candidates who are nominated for a seat on both Steering and CoCC within the
+  same 12-month period must:
+  - (if elected) accept a seat on the committee whose election concludes first
+  - be withdrawn from candidacy in second election to conclude
+- Standing Steering or CoCC members must have served on their committee for a
+  period of twelve (12) months to be considered eligible for candidacy in the
+  other committee's election
```
Comment (qedrakmar) [inline elections/committee-conflict-of-interest.md:53]: I like this sentiment, but I see a problem with the number 12 months, specifically in how the current elections are lined up.  If the elections were staggered by 6 months, this would be completely reasonable, but if the elections are close together, as they are right now, and they happen at about the same time every year, this places an unbalanced restriction on the committee that has its elections first.  Whatever the candidates reasons might be for wanting to change committees in the first place (for example: wanting more/less emotional labor), I would suggest this number change to slightly less (10 or 11 months), to not be such a unidirectional disadvantage.  This would also avoid even the perception of any possible artificial forcing function pushing back the annual cycle of elections.

Comment (qedrakmar) [reply elections/committee-conflict-of-interest.md:53]: As I'm thinking about this, we could also say slightly longer (14 months), and still avoid the imbalance when elections are close on the calendar.

File: elections/committee-conflict-of-interest.md [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+- Kubernetes Community members who have recently voluntarily vacated a seat on either
+  Steering or CoCC without serving the complete term will be ineligible for candidacy for a period of three (3)
+  months
```
Comment (cblecker) [inline elections/committee-conflict-of-interest.md:56]: So I'm going to challenge this a little bit, so I'm going to tag the folks who were suggesting something like this as I'd like their opinion (cc @katcosgrove @celestehorgan)

While I completely understand that we don't want folks jumping from committee to committee on a regular basis for reasons of continuity and burden on their fellow committee members, I fear that putting these kinds of hard restrictions on the election processes will have the possible unwanted side effects:
- Folks may choose not to run for a committee that they wish to serve on, because they might also hypothetically want to run in a different election within the 12 months if they are elected. This would have the effect of shrinking the possible candidate pools.
- These are unpaid, voluntary positions. Forcing someone to serve in a capacity that they no longer wish to is not only unfair to the individual, but it's unfair to the community as a whole who benefits from their service.
- Employer support for upstream work can change. While an employer may provide time to serve on one committee, they may not provide that time to serve on the other.. this again would hold someone in a position where they don't want to be.

I think in practicality, some of this could be reduced by having a greater staggering in the elections.. right now we have CoCC in August and Steering in September. If they were staggered more, then we wouldn't have any risk of concurrent election periods and less possibility of thrashing if folks move between committees. But having these long, forced ineligibility periods I feel puts strain on individuals who are serving the community in really important ways.

Very open to other suggestions that might better solve these issues.

**Review by cblecker — CHANGES_REQUESTED** [review]

File: elections/committee-conflict-of-interest.md (hunk 12 above)
Comment (aojea) [reply elections/committee-conflict-of-interest.md:56]: just curious, what is the spirit of this norm and why 3 months? it seems that it tries to avoid some specific thing to happen but I can´t understand what

Comment (katcosgrove) [reply elections/committee-conflict-of-interest.md:56]: @cblecker:

Our intent with this phrasing was that no potential for conflict exists until a committee position is actually held, so a candidate would be able to run for two positions within a 12-month period assuming they were not successful in their first attempt. If that's unclear, I'm happy to be more explicit!

The 3-month timeout period after voluntarily leaving your seat is there to discourage people from doing that simply to attempt to campaign for a seat on a more professionally lucrative committee. Allowing for committee hopping is especially problematic for the CoCC, where losing one member early is significantly more impactful than it is for Steering due to the much smaller committee size, but it's unfair to Steering as well and to the community at large due to the instability it causes. However, if we think the risk involved in voluntarily stepping down to run in another election the candidate may lose is deterrent enough without the 3-month timeout, I won't fight too hard to keep it. The individual should still be required to have served 12 months of their existing term before being eligible to run for a seat on another committee without first stepping down, though.

Staggering the elections more would certainly help solve this issue, but how doable is that in actual practice?
